### PR TITLE
feat: impl raw-provider and indexer-collector

### PR DESCRIPTION
**Pull request by huwenchao**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

reference: https://talk.nervos.org/t/proposal-with-a-50-000-ckb-bonus-for-adapting-pw-core-to-back-end-usage/5301

### CODE REVIEW

File: src/signers/ckb-signer.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  public verify = (message: string | Buffer, sig: string | Buffer) => {
+    const msg = typeof message === 'string' ? hexToBytes(message) : message;
+    const signature = typeof sig === 'string' ? hexToBytes(sig) : sig;
+    return this.key.verify(msg, signature as any);
+  }
+
+  public signRecoverable = (message: string | Uint8Array): string => {
+    const msg = typeof message === 'string' ? hexToBytes(message) : message;
```
Comment (timfaner) [inline src/signers/ckb-signer.ts:95]: need to use blake2b to hash the message, then sign the message

Comment (timfaner) [reply src/signers/ckb-signer.ts:95]: Maybe using this one https://github.com/lay2dev/pw-core/blob/861310b3dd8638f668db1a08d4c627db4c34d815/src/hashers/blake2b-hasher.ts#L16

Comment (huwenchao) [reply src/signers/ckb-signer.ts]: The message is already  hashed when invoke signRecoverable.

**Comment by huwenchao** [comment]
This PR is finished. It is tested in AGGRON Testnet with the example script in `scripts` folder.

**Pull request merged**